### PR TITLE
Upgrade typescript to latest version, and fix new errors

### PR DESCRIPTION
**Pull request by adamabeshouse**

### CODE REVIEW

File: src/pages/resultsView/ResultsViewPageStore.ts [hunk 1]
```diff
@@ -273,8 +273,10 @@ export enum SampleListCategoryType {
 }
 
 export enum GeneticEntityType {
-    'GENE' = 'gene',
-    'GENESET' = 'geneset',
+    'GENE' = 'GENE',
```
Comment (adamabeshouse) [inline src/pages/resultsView/ResultsViewPageStore.ts:276]: @dippindots you worked on geneset stuff right? could you take a look at these changes I made relating to GeneticEntityType and make sure I haven't done anything weird?

Comment (dippindots) [reply src/pages/resultsView/ResultsViewPageStore.ts:276]: @adamabeshouse This looks good to me! It's in sync with the backend.

File: src/pages/resultsView/enrichments/EnrichmentsUtil.tsx [hunk 2]
```diff
@@ -611,7 +611,7 @@ export function getAlterationEnrichmentColumns(
         enrichedGroupColum.tooltip = (
             <table>
                 <tr>
-                    <td>Log ratio > 0</td>
+                    <td>Log ratio {'>'} 0</td>
```
Comment (adamabeshouse) [inline src/pages/resultsView/enrichments/EnrichmentsUtil.tsx:614]: officially, JSX does not allow >, <, {, } as plaintext characters, and the new version of typescript catches that

File: src/pages/staticPages/tools/oncoprinter/OncoprinterGeneticUtils.ts [hunk 3]
```diff
@@ -707,8 +710,10 @@ export function annotateGeneticTrackData(
             const newObj = _.clone(object);
             newObj.data = newObj.data.filter(d => {
                 // clear previous annotations
-                delete d.oncoKbOncogenic;
-                delete d.putativeDriver;
+                delete (d as Partial<OncoprinterGeneticTrackDatum_Data>)
```
Comment (adamabeshouse) [inline src/pages/staticPages/tools/oncoprinter/OncoprinterGeneticUtils.ts:713]: the new typescript has a check which does not allow you to use `delete` on a member which is not potentially undefined in the type

File: src/shared/lib/comparison/ComparisonStore.ts [hunk 4]
```diff
@@ -170,6 +170,13 @@ export default class ComparisonStore {
     protected async saveAndGoToSession(newSession: Session) {
         throw new Error(`saveAndGoToSession must be implemented in subclass`);
     }
+    abstract _session: MobxPromise<Session>;
+    abstract _originalGroups: MobxPromise<ComparisonGroup[]>;
+    abstract overlapStrategy: OverlapStrategy;
+    abstract usePatientLevelEnrichments: boolean;
```
Comment (adamabeshouse) [inline src/shared/lib/comparison/ComparisonStore.ts:176]: the new typescript catches if the superclass declares an instance property and the subclass overrides it using a `get` accessor, so after some difficulty I realized that the solution is to declare these as abstract properties

File: src/shared/components/GeneSelectionBox/GeneSymbolValidatorMessage.tsx [hunk 5]
```diff
@@ -246,10 +246,7 @@ export default class GeneSymbolValidatorMessage extends React.Component<
 
         return (
             <div id="geneBoxValidationStatus">
-                <GeneSymbolValidatorMessageChild
-                    replaceGene={this.props.replaceGene}
-                    {...this.props}
-                />
+                <GeneSymbolValidatorMessageChild {...this.props} />
```
Comment (alisman) [inline src/shared/components/GeneSelectionBox/GeneSymbolValidatorMessage.tsx:249]: what happened to replaceGene?

File: src/shared/lib/comparison/ComparisonStore.ts (hunk 4 above)
Comment (alisman) [reply src/shared/lib/comparison/ComparisonStore.ts:176]: abstract forces the subclass to define it?

**Review by alisman — APPROVED** [review]

File: src/shared/components/GeneSelectionBox/GeneSymbolValidatorMessage.tsx (hunk 5 above)
Comment (adamabeshouse) [reply src/shared/components/GeneSelectionBox/GeneSymbolValidatorMessage.tsx:249]: it's already in `this.props`

File: src/shared/lib/comparison/ComparisonStore.ts (hunk 4 above)
Comment (adamabeshouse) [reply src/shared/lib/comparison/ComparisonStore.ts:176]: I believe so, yes. and it allows the property to be implemented using an instance property or a getter/setter

**Pull request merged**